src/ResourceManager/Sql/Commands.Sql/Common/AzureSqlCmdletBaseBase.cs

```diff
@@ -0,0 +1,169 @@
+                var serverName = (serverProperty == null) ? string.Empty : serverProperty.GetValue(m).ToString();
+
+                var databaseProperty = m.GetType().GetProperty("DatabaseName");
+                var databaseName = (databaseProperty == null) ? string.Empty : databaseProperty.GetValue(m).ToString();
+
+                if (!string.IsNullOrEmpty(serverName))
+                {
+                    if (!string.IsNullOrEmpty(databaseName))
+                    {
+                        return string.Format("{0}.{1}", serverName, databaseName);
+                    }
+                    return serverName;
+                }
+                if (!string.IsNullOrEmpty(databaseName))
+                {
+                    return databaseName;
+                }
+            }
+
+            return string.Empty;
+        }
+
+        /// <summary>
+        /// The ModelAdapter object used by this cmdlet
+        /// </summary>
+        public A ModelAdapter { get; internal set; }
+
+        /// <summary>
+        /// Gets an entity from the service
+        /// </summary>
+        /// <returns>A model object</returns>
+        protected abstract M GetEntity();
+
+        /// <summary>
+        /// Updates the given model element with the cmdlet specific operation 
+        /// </summary>
+        /// <param name="model">A model object</param>
+        protected virtual M ApplyUserInputToModel(M model) { return model; }
+
+        /// <summary>
+        /// This method is responsible to call the right API in the communication layer that will eventually send the information in the 
+        /// object to the REST endpoint
+        /// </summary>
+        /// <param name="entity">The model object with the data to be sent to the REST endpoints</param>
+        protected virtual M PersistChanges(M entity) { return default(M); }
+
+        /// <summary>
+        /// Returns true if the model object that was constructed by this cmdlet should be written out
+        /// </summary>
+        /// <returns>True if the model object should be written out, False otherwise</returns>
+        protected virtual bool WriteResult() { return true; }
+
+        /// <summary>
+        /// Creation and initialization of the ModelAdapter object
+        /// </summary>
+        /// <param name="subscription">The AzureSubscription in which the current execution is performed</param>
+        /// <returns>An initialized and ready to use ModelAdapter object</returns>
+        protected abstract A InitModelAdapter(IAzureSubscription subscription);
+
+        /// <summary>
+        /// Transforms the given model object to be an object that is written out
+        /// </summary>
+        /// <param name="model">The about to be written model object</param>
+        /// <returns>The prepared object to be written out</returns>
+        protected virtual object TransformModelToOutputObject(M model)
```

Comment:
This is only used in a couple places.
[Get-AzureRmSqlDatabaseDataMaskingRule](https://github.com/Azure/azure-powershell/blob/preview/src/ResourceManager/Sql/Commands.Sql/Data%20Masking/Cmdlet/GetAzureSqlDatabaseDataMaskingRule.cs#L50) cmdlet uses it to filter the results before showing the user.
[SqlDatabaseDataMaskingRuleCmdletBase](https://github.com/Azure/azure-powershell/blob/preview/src/ResourceManager/Sql/Commands.Sql/Data%20Masking/Cmdlet/SqlDatabaseDataMaskingRuleCmdletBase.cs#L63) also uses it to do client side filtering of the results.
[New-AzureRmSqlDatabase](https://github.com/Azure/azure-powershell/blob/preview/src/ResourceManager/Sql/Commands.Sql/Database/Cmdlet/NewAzureSqlDatabase.cs#L203) Actually uses it to modify the response slightly.

That being said, all this code is pre-existing code that already existed. I'm just moving it to a new file.  I might have a different way to achieve my main goal without moving this stuff around.